src/main/java/de/unistuttgart/overworldbackend/service/CourseService.java

```diff
@@ -143,4 +148,87 @@ private Dungeon configureDungeon(int dungeonId, DungeonConfig dungeonConfig) {
     }
     return new Dungeon(dungeonConfig.getStaticName(), "", false, minigames, npcs, dungeonId);
   }
+
+  public CourseDTO cloneCourse(int id, CourseInitialData courseInitialData) {
+    Course course = courseRepository
+      .findById(id)
+      .orElseThrow(() ->
+        new ResponseStatusException(HttpStatus.NOT_FOUND, String.format("There is no course with id %s.", id))
+      );
+    List<World> worlds = new ArrayList<>();
+    course.getWorlds().forEach(world -> cloneWorld(world, worlds));
+
+    Course cloneCourse = new Course(
+      courseInitialData.getCourseName(),
+      courseInitialData.getSemester(),
+      courseInitialData.getDescription(),
+      false,
+      worlds
```

Comment:
```suggestion

    Course cloneCourse = new Course(
      courseInitialData.getCourseName(),
      courseInitialData.getSemester(),
      courseInitialData.getDescription(),
      false,
      course.getWorlds().parallelStream().map(this::cloneWorld).collect(Collectors.toCollection(ArrayList::new))
```
Also, the `false` is `isActive`, right?
Perhaps you should define a constant `DEFAULT_IS_ACTIVE` so that we can easily switch it, as it is already intended.